src/ceph-disk/ceph_disk/main.py

```diff
@@ -2123,20 +2179,89 @@ def __init__(self, args):
         super(PrepareBluestoreBlock, self).__init__(args)
 
     def get_space_size(self):
-        return 0  # get as much space as possible
+        block_size = get_conf(
+            cluster = self.args.cluster,
+            variable = 'bluestore_block_size',
+        )
 
-    def desired_partition_number(self):
-        if self.args.block == self.args.data:
-            num = 2
+        if block_size is None:
+            return 0  # get as much space as possible
         else:
-            num = 0
-        return num
+            return int(block_size) / 1048576 # MB
+
+    def desired_partition_number(self):
+        # FIXME: use a fixed partition number for now
```

Comment:
I think it's just simpler and more predictable from a user point of view. Do you see a reason why it should be fixed later on ? Initially I had the  `if self.args.block == self.args.data:` because it mimics what is done with journals. So the question really is: does it make sense for bluestore to have self.args.block* on a separate device ? If so and if there can be more than one such partition, the partition number for all of these need to be dynamic.